File: Fun/Chess/NullPiece.java
package Chess;

public class NullPiece extends ChessPiece {
    public NullPiece(char player) {
        super(player);
    }

    @Override
    public boolean isValidMove(int startRow, int startCol, int endRow, int endCol, ChessBoard board) {
        return false;
    }
    public String toString() {
        return ".";
    }
    public String getType() {
        return "N";
    }
    public int getValue() {
        return 0;
    }
    
}

File: site/public/Chess-js/NullPiece.js
class NullPiece extends ChessPiece {
    constructor(color) {
        super(color);
    }

    isValidMove(startRow, startCol, endRow, endCol, board) {
        return false;
    }

    toString() {
        return ".";
    }

    getType() {
        return "N";
    }

    getValue() {
        return 0;
    }
}